app/models/user.rb

```diff
@@ -14,14 +14,15 @@ class User < ActiveRecord::Base
     :twitter_username
   ].freeze
 
+  before_destroy :yank_gems
```

Comment:
[Previous discussion](https://github.com/rubygems/rubygems.org/pull/1301#discussion_r69366108) was marked outdated diff on rebase.

It is to be decided whether single owner gems should be yanked on profile delete or a new [rule](https://github.com/rubygems/rubygems.org/blob/master/app/models/pusher.rb#L23-L24) should be added for pushing to orphaned gems.

If we don't yank gem on profile delete and add rule for pushing to orphaned gems, what happens to older versions of orphaned gem? Does new owner becomes owner of previous versions or we yank older versions when the gem is adopted or something else.